src/language_features/semantic_highlighting.rs

```diff
@@ -91,18 +91,16 @@ pub fn editor_update(meta: EditorMeta, params: EditorParams, ctx: &mut Context)
 }
 
 pub fn debug_scopes(meta: EditorMeta, ctx: &mut Context) {
-    let semhl = ctx
+    let scopes = ctx
         .capabilities
         .as_ref()
-        .unwrap()
-        .semantic_highlighting
-        .as_ref();
-    semhl.map(|sh| {
-        if sh.scopes.is_some() {
-            let command = format!("echo -debug %§{:?}§", sh.scopes.as_ref().unwrap());
-            ctx.exec(meta, command.to_string());
-        }
-    });
+        .and_then(|x| x.semantic_highlighting.as_ref())
+        .and_then(|x| x.scopes.as_ref());
+
+    if let Some(scopes) = scopes {
+        let command = format!("echo -debug %§{:?}§", scopes);
```

Comment:
I'm not sure if logic to debug scopes here is correct, because invoking `lsp-semantic-available-scopes` as suggested [here](https://github.com/kak-lsp/kak-lsp/blob/34e888eea109ad1b2fbaee3b0b7afab46b38f1c8/kak-lsp.toml#L10) never printed anything to the debug buffer at least with rust-analyzer and gopls.

Reply:
Don't they use the new semantic tokens api? cause the available scopes is for the old semantic highlighting api iirc

Reply:
> Don't they use the new semantic tokens api?

Can't answer, just wanted to point out that calling `lsp-semantic-available-scopes` doesn't do anything useful.
Also I can't tell what is difference between `[semantic_scopes]` and `[semantic_tokens]`. Former is referring to texmate scopes.

Reply:
Yeah it's probably lacking some documentation...